tensorflow/compiler/tests/tensor_array_ops_test.py

```diff
@@ -325,21 +325,24 @@ def testTensorArrayWriteWrongIndexOrDataTypeFails(self):
         ta.write(-1, np.int32(7)).flow.eval()
 
   def testTensorArrayReadWrongIndexOrDataTypeFails(self):
-    with self.test_session(), self.test_scope():
-      ta = tensor_array_ops.TensorArray(
-          dtype=dtypes.float32, tensor_array_name="foo", size=3)
-
-      w0 = ta.write(0, [[4.0, 5.0]])
-
-      # Test reading wrong datatype
-      r0_bad = gen_data_flow_ops._tensor_array_read_v3(
-          handle=w0.handle, index=0, dtype=dtypes.float64, flow_in=w0.flow)
-      with self.assertRaisesOpError(
-          "TensorArray dtype is float but op has dtype double."):
-        r0_bad.eval()
-
-      # Test reading from a different index than the one we wrote to
-      w0.read(1)
+    if len(self.float_types) > 1:
+      dtype1 = self.float_types[0]
```

Comment:
What it is really doing is picking 2 float types which are not the same. In the case of a CPU or GPU it will probably pick f32 and f64. However for our device it is f16 and f32, since we don't have f64 support (yet). 

I suspect the types selected depend on the order that the test framework specifies them. 

Perhaps a good comment would be to describe the above (in a concise way)